src/shared_modules/utils/rocksDBIterator.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 9, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _ROCKS_DB_ITERATOR_HPP
+#define _ROCKS_DB_ITERATOR_HPP
+
+#include "rocksdb/iterator.h"
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+class RocksDBIterator
+{
+public:
+    // End iterator
+    RocksDBIterator()
```

Comment:
Is the default constructor needed? Does this class apply the rule of [zero/three/five](https://en.cppreference.com/w/cpp/language/rule_of_three) to provide movement semantics?

Reply:
It needs to be explicit(default constructor) because the begin method needs explicit declaration.